client/docs/plugin_model.md

```diff
@@ -0,0 +1,169 @@
+<!---
+Licensed under Creative Commons Attribution 4.0 International License
+https://creativecommons.org/licenses/by/4.0/
+--->
+
+# Building Contract Plugins #
+
+The plugins module provides a means to build and use operations that
+is independent (as much as possible) of the actual invocation of the
+operation. For example, the `mock-contract` plugin wraps the
+`inc_value` method in the contract object with an interface so that it
+can be invoked as part of a Python script, a pdo-shell script, or a
+bash script. Similarly, the `mint_tokens` command may invoke a series
+of operations across multiple contract objects in order to mint new
+NFT tokens.
+
+There are three general styles of plugins: *contract operations*,
+*contract commands* and *scripts*. Loosely, a contract operation wraps
+the invocation of a single method on a contract object. Contract
+operations are parameterized to interact with a specific contract
+object with defaults for enclave service and ledger
+connections. Contract commands provide a logical operation that spans
+multiple invocations (possibly across multiple contract
+objects). Scripts simply provide a useful local function (such as
+updating the current configuration). The PDO client library includes a
+basic set of scripts that should be sufficient for most uses.
```

Comment:
Presentation: This would be a little more reader-friendly if the description of each plugin style was in its own bullet point, just to break up the larger block of text.

Reply:
Updated.